includes/widgets.php

```diff
@@ -41,7 +41,7 @@ function __construct() {
 	/** @see WP_Widget::widget */
 	function widget( $args, $instance ) {
 
-		if ( isset( $instance['hide_on_checkout'] ) && edd_is_checkout() ) {
+		if ( isset( $instance['hide_on_checkout'] ) && $instance['hide_on_checkout'] && edd_is_checkout() ) {
```

Comment:
This could be simplified to just `if ( ! empty( $instance['hide_on_checkout'] ) `